api/tests/unit/domain/usecases/get-campaign-participations-with-results_test.js

```diff
@@ -0,0 +1,102 @@
+const { expect, sinon, catchErr } = require('../../../test-helper');
+const getCampaignParticipationsWithResults = require('../../../../lib/domain/usecases/get-campaign-participations-with-results');
+const { UserNotAuthorizedToAccessEntity } = require('../../../../lib/domain/errors');
+
+describe('Unit | UseCase | get-campaign-participations-with-results', () => {
+
+  let campaignParticipationsResult;
+  let requestErr;
+
+  let options;
+
+  const userId = Symbol('user id');
+  const assessmentId = Symbol('assessment id');
+  const sharedAt = Symbol('shared at');
+  const campaignId = Symbol('campaign id');
+  const targetProfile = Symbol('target profile');
+  const competences = Symbol('competences');
+  const assessment = Symbol('assessment');
+  const knowledgeElements = Symbol('knowledge elements');
+  const campaignParticipation = {
+    assessmentId, userId, sharedAt, campaignId,
+    addCampaignParticipationResult: sinon.stub()
+  };
+  const campaignParticipations = { models: [ campaignParticipation ] };
+  const campaignParticipationsWithResults = { models: [ campaignParticipation ] };
+
+  const campaignRepository = { checkIfUserOrganizationHasAccessToCampaign: sinon.stub() };
+  const campaignParticipationRepository = { findWithUsersPaginated: sinon.stub() };
+  const targetProfileRepository = { getByCampaignId: sinon.stub() };
+  const competenceRepository = { list: sinon.stub() };
+  const assessmentRepository = { get: sinon.stub() };
+  const smartPlacementKnowledgeElementRepository = { findUniqByUserId: sinon.stub() };
+
+  beforeEach(() => {
+    options = null;
+    requestErr = null;
+  });
+
+  context('the campaign participation is searched by campaign id', () => {
+    beforeEach(() => {
+      options = { filter: { campaignId } };
+    });
+    context('the assessment belongs to the user', () => {
+      beforeEach(async () => {
+        campaignRepository.checkIfUserOrganizationHasAccessToCampaign.resolves(true);
+        campaignParticipationRepository.findWithUsersPaginated.resolves(campaignParticipations);
+        targetProfileRepository.getByCampaignId.resolves(targetProfile);
+        competenceRepository.list.resolves(competences);
+        assessmentRepository.get.resolves(assessment);
+        smartPlacementKnowledgeElementRepository.findUniqByUserId.resolves(knowledgeElements);
+
+        campaignParticipationsResult = await getCampaignParticipationsWithResults({
+          userId,
+          options,
+          campaignParticipationRepository,
+          campaignRepository,
+          competenceRepository,
+          assessmentRepository,
+          targetProfileRepository,
+          smartPlacementKnowledgeElementRepository
+        });
+      });
+      it('should check if the user organization has access to the campaign', () => {
+        expect(campaignRepository.checkIfUserOrganizationHasAccessToCampaign).to.have.been.calledWithExactly(campaignId, userId);
```

Comment:
Suggestion : l'utilisation de `withArgs` sur les _stubs_ permet d'obtenir des tests plus légers que ces longues séries de `to.have.been.calledWithExactly` qui présentent l'inconvénient de reproduire exactement l'implémentation. Je veux bien faire une démo à l'occasion, sinon demander à @Radioreve 😉